docs/scripts/util/checkLink.ts

```diff
@@ -18,6 +18,9 @@ export type LinkInfo = {
 };
 const requestedUrl: Set<string> = new Set();
 
+const hostName = 'http://localhost:3000';
```

Comment:
Won't this fail when the docs are running in production mode on a different port?

Reply:
That's a good point, which was the reason why it used a regex to find the hostname.

However, link-checker only test against localhost in the reusable-e2e https://github.com/aws-amplify/amplify-ui/blob/00129b743a7c117e0c9a3bf7cb834203c1c855e5/.github/workflows/reusable-e2e.yml#L687, so 'http://localhost:3000' is good enough for our current use case.

Do you think we should handle more use cases?

Reply:
But surely we aren't running docs in dev mode in e2e tests? Do we have some config that's making it pick port 3000?

Reply:
BTW, the target site's hostName is set in the environment variable, which is 'http://localhost:3000' in Github

https://github.com/aws-amplify/amplify-ui/blob/1bf5b330d108fb4eee28f3951fd5b0a45b4547b0/docs/src/data/sitePaths.ts#L8

If we turn it on in Prod, we'll need to change `SITE_URL` in Prod env.

Reply:
If we want to test against Prod, I think we can have `Build and RunTime Test` to do that. So that'll be like

- PR test: localhost
- Build and RunTIme test: Prod "https://ui.docs.amplify.aws/"

Does that make sense?

Reply:
I was wrong about this, just tried locally and next.js prod build runs on port 3000, like you have.